x-pack/plugins/stack_connectors/public/components/text_area_with_autocomplete.tsx

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useMemo, useCallback, useEffect } from 'react';
+import getCaretCoordinates from 'textarea-caret';
+import { Properties } from 'csstype';
+import {
+  EuiTextArea,
+  EuiFormRow,
+  EuiSelectable,
+  EuiSelectableOption,
+  EuiPortal,
+  EuiHighlight,
+  EuiOutsideClickDetector,
+  useEuiTheme,
+  useEuiBackgroundColor,
+} from '@elastic/eui';
+import { ActionVariable } from '@kbn/alerting-plugin/common';
+import { AddMessageVariables } from '@kbn/alerts-ui-shared';
+import { filterSuggestions } from '../lib/filter_suggestions_for_autocomplete';
+import { templateActionVariable } from '../lib/template_action_variable';
+
+export interface TextAreaWithAutocompleteProps {
+  editAction: (property: string, value: any, index: number) => void;
+  errors?: string[];
+  index: number;
+  inputTargetValue?: string;
+  isDisabled?: boolean;
+  label: string;
+  messageVariables?: ActionVariable[];
+  paramsProperty: string;
+}
+const selectableListProps = { className: 'euiSelectableMsgAutoComplete' };
+
+export const TextAreaWithAutocomplete: React.FunctionComponent<TextAreaWithAutocompleteProps> = ({
+  editAction,
+  errors,
+  index,
+  inputTargetValue,
+  isDisabled = false,
+  label,
+  messageVariables,
+  paramsProperty,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const backgroundColor = useEuiBackgroundColor('plain');
+
+  const textAreaRef = React.useRef<HTMLTextAreaElement | null>(null);
+  const selectableRef = React.useRef<EuiSelectable | null>(null);
```

Comment:
I see. In that case I can skip it and does not have errors.
But I see some article which suggested to it with null because it's more explicit.
What do you think?
```
It is also a good practice to explicitly declare the type of useRef as shown in below code snippet. In this case, it is null | HTMLParagraphElement.
```

Reply:
I think I get what you mean. 
I've used null, because it seems that HTMLTextAreaElement type recognise it as a part of this type. If I change it to undefined, it will be a type error for EuiTextArea component. It needs time to investigate how to fix it.
Can we stick in that case to null to save time? It works almost the same as undefined.